source/extensions/common/wasm/wasm_vm.h

```diff
@@ -287,7 +291,7 @@ struct SaveRestoreContext {
 };
 
 // Create a new low-level WASM VM using runtime of the given type (e.g. "envoy.wasm.runtime.wavm").
-WasmVmPtr createWasmVm(absl::string_view runtime);
+WasmVmPtr createWasmVm(absl::string_view runtime, Stats::ScopeSharedPtr scope);
```

Comment:
nit: pass either by const-ref or by explicit `&&` all the way through for clarity.

Reply:
FYI: I checked the compiler output and there is no performance benefit to using std::move + && if the variable goes out of scope immediately as the compiler does copy elision.  Since the pointer is shared there is not ownership being passed so I don't understand what you mean by "clarity" here as we typically pass with && to indicate that there is a passing of ownership which is not the case here.  However, taking your suggestions:

Using && means that the caller will have to make an explicit copy if the variable is still in use after the call or std::move if it is not which clutters up the code and adds an extra burden as the caller who now has to manage the life cycle of what is logically a shared object. If the caller decides to take the easy way out and always make a copy, the code will be less efficient as the copy is explicit and can't be removed via copy elision by the compiler.

Switching to const-ref.  That should produce equally efficient code and adds no burden to the caller, although because there is now non-local sharing (i.e. the variable in the caller is now aliased into the callees) it is nominally less safe (an indirect function call from the callee which causes the caller to clear or or replace that variable could effect the variable in the callee since it is a reference).  Personally I think it would be better to pass by value here since it is less code, just as efficient and safer (no non-local sharing).

All that said, I switched it to const-ref since that seems to be the more common idiom in Envoy when dealing with shared objects and no passing of ownership. 

PTAL